subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -756,25 +805,24 @@ static int tls_set_credential(struct tls_context *tls,
 			      struct tls_credential *cred)
 {
 	switch (cred->type) {
+
 	case TLS_CREDENTIAL_CA_CERTIFICATE:
 		return tls_add_ca_certificate(tls, cred);
 
 	case TLS_CREDENTIAL_SERVER_CERTIFICATE:
+		return tls_add_own_cert(tls, cred);
+
+	case TLS_CREDENTIAL_PRIVATE_KEY:
 	{
 		struct tls_credential *priv_key =
```

Comment:
Not sure I follow this one, can't we just set `tls_set_private_key(tls, cred)`? The currently processed credential is the private key, I don't see a reason to look it up again.